src/libraries/System.Net.Quic/src/System/Net/Quic/Implementations/MsQuic/MsQuicStream.cs

```diff
@@ -330,7 +330,8 @@ private CancellationTokenRegistration SetupWriteStartState(bool emptyBuffer, Can
             {
                 throw new InvalidOperationException(SR.net_quic_writing_notallowed);
             }
-            if (_state.SendState == SendState.Aborted)
+            // Use Volatile.Read to ensure we read the actual SendErrorCode set by the racing callback thread.
```

Comment:
"callback thread" in this case means MsQuic calling into the managed code?

Reply:
yes

Reply:
There's a separate race condition here: #69491